src/sentry/integrations/slack/views/link_identity.py

```diff
@@ -40,33 +46,72 @@ class SlackLinkIdentityView(BaseView):
     Django view for linking user to slack account. Creates an entry on Identity table.
     """
 
+    _METRICS_SUCCESS_KEY = "sentry.integrations.slack.link_identity_view.success"
+    _METRICS_FAILURE_KEY = "sentry.integrations.slack.link_identity_view.failure"
+
     @method_decorator(never_cache)
-    def handle(self, request: Request, signed_params: str) -> HttpResponse:
+    def dispatch(self, request: HttpRequest, *args, **kwargs) -> HttpResponseBase:
         try:
+            signed_params = kwargs.pop("signed_params")
             params = unsign(signed_params)
-        except (SignatureExpired, BadSignature):
+        except (SignatureExpired, BadSignature) as e:
+            _logger.warning("dispatch.signature_error", exc_info=e)
+            metrics.incr(self._METRICS_FAILURE_KEY, tags={"error": str(e)}, sample_rate=1.0)
             return render_to_response(
                 "sentry/integrations/slack/expired-link.html",
                 request=request,
             )
 
-        organization, integration, idp = get_identity_or_404(
-            ExternalProviders.SLACK,
-            request.user,
-            integration_id=params["integration_id"],
+        try:
+            organization, integration, idp = get_identity_or_404(
+                ExternalProviders.SLACK,
+                request.user,
+                integration_id=params["integration_id"],
+            )
+        except Http404:
+            _logger.exception(
+                "get_identity_error", extra={"integration_id": params["integration_id"]}
+            )
+            metrics.incr(self._METRICS_FAILURE_KEY + ".get_identity", sample_rate=1.0)
+            raise
+
+        _logger.info("get_identity_success", extra={"integration_id": params["integration_id"]})
+        metrics.incr(self._METRICS_SUCCESS_KEY + ".get_identity", sample_rate=1.0)
+        params.update({"organization": organization, "integration": integration, "idp": idp})
+        return super().dispatch(request, params=params)
+
+    def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        params = kwargs["params"]
+        organization, integration = params["organization"], params["integration"]
+
+        return render_to_response(
+            "sentry/auth-link-identity.html",
+            request=request,
+            context={"organization": organization, "provider": integration.get_provider()},
+        )
+
+    def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        params = kwargs["params"]
+        organization, integration, idp = (
+            params["organization"],
+            params["integration"],
+            params["idp"],
```

Comment:
are we doing schema/contract checks anywhere and returning the proper error code?

Reply:
Can we use a serializer or a dataclass and set this up to be contract based?